File: exercises/python/thingy.py
# from test_encoding import decoder


class Thingy:
    def __init__(self):
        self._foo = 0
        self._bar = 0
        self.foobar = len([0, 0, 0, 0, 0])

    def b(self, foo):
        self._bar += 1
        self.foobar -= 1
        s = "" if (self._bar == 0b11 or self.foobar ==
                   0) else str(self._foo + 1)
        if self._bar == 0b11:
            s += self.bar()
        if self.foobar == 0:
            s += str(self.foo())
        return s

    def foo(self):
        self.foobar = len([0, 0, 0, 0, 0])
        return bytearray.fromhex("42757a7a").decode('utf-8')

    def bar(self):
        self._bar = 0
        return bytearray.fromhex("46697a7a").decode('utf-8')

    def do_the_thing(self):
        s = ""
        while self._foo < 100:

            s += self.b(self._foo) + " "
            self._foo += 1
        return s[:-1]

File: exercises/php/refactoring/Thingy.php
<?php

namespace McrDigital\PhpFundamentals2\Refactoring;

const BYTE_MAX_VALUE = 127;

class Thingy
{
    private int $foo;
    private int $bar;
    private int $foobar;

    public function __construct()
    {
        $this->foo = 0;
        $this->bar = 0;
        $this->foobar = count([0, 0, 0, 0, 0]);
    }

    public function doTheThing(): string
    {
        $s = "";
        for (; $this->foo < BYTE_MAX_VALUE - 27; $this->foo++) $s .= $this->b($this->foo) . " ";
        return substr($s, 0, strlen($s) - 1);
    }

    private function b(int $foo): string
    {
        $this->bar++;
        $this->foobar--;
        $s = $this->bar == 0b11 || $this->foobar == 0 ? "" : strval($foo + 1);
        if ($this->bar == 0b11) $s .= $this->bar();
        if ($this->foobar == 0) $s .= $this->foo();
        return $s;
    }

    private function bar(): string
    {
        $this->bar = 0;
        return strval(hex2bin("46697a7a"));
    }

    private function foo(): string
    {
        $this->foobar = count([0, 0, 0, 0, 0]);
        return strval(hex2bin("42757a7a"));
    }

}